feat: GEDI CalVal COPC

jjfrench opened the pull request:
- Add GEDI CalVal COPC dataset pipeline
- Remove non-MAAP pipelines

Git seems to be having a hard time with it - but I just copied the files into the gedi_calval_copc directory and renamed a couple of things

jjfrench committed:
feat: gedi_calval_copc pipeline

emileten commented:
Looks good to me @jjfrench, did you get your tests to pass?

jjfrench commented:
@emileten the tests did pass 👍🏼

emileten commented on stactools_pipelines/pipelines/gedi_calval_copc/config.yaml:
```diff
@@ -0,0 +1,7 @@
+  id: "gedi-calval-copc"
+  compute: "awslambda"
+  secret_arn: "arn:aws:secretsmanager:us-west-2:916098889494:secret:MAAP-STAC-auth-dev/MAAP-workflows-EsykqB" # cognito secret ARN
+  ingestor_url: "https://stac-ingestor.maap-project.org"
```
suggestion : change this to the 'test' ingestor (that ingests to the test catalog) for a first run to check if things work properly (maybe also shorten the csv for that test, since you don't need all the data to be ingested for the test)

jjfrench replied:
Good point! I'll make those changes

jjfrench committed:
test: DIT ingest test

emileten reviewed (commented):
Have you tried deploying this test version @jjfrench ? (Not telling you to do so, just that we should do it before merging. I can also try)

jjfrench committed:
chore: broken dockerfile push

jjfrench committed:
chore: pipeline specific granule dockerfile

jjfrench committed:
style: missing comma

jjfrench committed:
chore: create_item permissions to generate cloud-optimized data

jjfrench committed:
fix: updated docker image builds

jjfrench commented:
Collection is published to dev catalog, these changes reflect the necessary fixes to get it deployed:

Number of other fixes:
- pipeline granule dockerfile au lieu of root granule dockerfile
- upgraded aws-cdk-lib to latest version to update default nodejs runtime in cdk stack
- cdk inventory always either OUTPUT_LOCATION and INVENTORY_LOCATION or FILE_LIST
- cdk inventory correctly builds, process_inventory_chunk no longer referenced before creation
- cdk inventory using PYTHON_3_9 for athena table creator
- cdk pipeline function, added permissions for create_item to create cloud-optimized versions
- cdk pipeline function, added HOME for libraries generating cloud-optimized versions (PDAL)

emileten reviewed (approved):
Thanks for the fixes @jjfrench

omshinde reviewed (approved):
Thanks @jjfrench
